### PR TITLE
SIM-Transaction indexing problem

### PR DESCRIPTION
**Pull request by AgustinRamiroDiaz**
Fixes https://github.com/yeagerai/genlayer-simulator/issues/303

# What

- Does the necessary indexing in an array

# Why

- to fix a bug while deploying a contract

# Testing done

- manually created contracts
- ran e2e tests locally

# Checks

- [x] I have tested this code
- [x] I have reviewed my own PR
- [x] I have created an issue for this PR
- [x] I have set the PR name to the issue name

# User facing release notes

Fixed a bug when deploying a smart contract

### CODE REVIEW

File: .github/workflows/backend_integration_tests_pr.yml [hunk 1]
```diff
@@ -2,17 +2,16 @@ name: CI
...
       - opened
+      - ready_for_review
   pull_request_review:
     types:
       - submitted
 
 jobs:
   test:
-    if: ${{ github.event.review.state == 'APPROVED' || github.event_name == 'pull_request' }}
+    if: ${{ github.event.review.state == 'APPROVED' || (github.event_name == 'pull_request' && github.base_ref) }}
```
Comment (AgustinRamiroDiaz) [inline .github/workflows/backend_integration_tests_pr.yml:14]: Also fixed this to actually trigger the tests (`branches` refers to the source branch and not the target one)

**Review by bradleySuira — APPROVED** [review]
LGTM

File: .github/workflows/backend_integration_tests_pr.yml (hunk 1 above)
Comment (cristiam86) [reply .github/workflows/backend_integration_tests_pr.yml:14]: @AgustinRamiroDiaz shouldn't it be compared to some branch name?

File: backend/database_handler/transactions_processor.py [hunk 2]
```diff
@@ -69,9 +69,9 @@ def insert_transaction(
     def get_transaction_by_id(self, transaction_id: int) -> dict:
         condition = f"id = {transaction_id}"
         transaction_data = self.db_client.get(self.db_transactions_table, condition)
-        if(len(transaction_data) == 0):
+        if len(transaction_data) == 0:
             return None
-        return self._parse_transaction_data(transaction_data)
+        return self._parse_transaction_data(transaction_data[0])
```
Comment (cristiam86) [inline backend/database_handler/transactions_processor.py:74]: @AgustinRamiroDiaz this is already on main. You can see it here: https://github.com/yeagerai/genlayer-simulator/blob/main/backend/database_handler/transactions_processor.py#L74

**Review by cristiam86 — CHANGES_REQUESTED** [review]
Please point the PR to staging

File: .github/workflows/backend_integration_tests_pr.yml (hunk 1 above)
Comment (AgustinRamiroDiaz) [reply .github/workflows/backend_integration_tests_pr.yml:14]: You are right! Fixed

**Comment by AgustinRamiroDiaz** [comment]
> @AgustinRamiroDiaz please note that the index change is already on staging so you probably need to merge from it first

@cristiam86 I don't see it currently in staging

https://github.com/yeagerai/genlayer-simulator/blob/66225cb90a6ae14956cbc2e23dc19ccd73b80f8a/backend/database_handler/transactions_processor.py#L74

**Pull request merged**